service/history/historyEngine.go

```diff
@@ -1232,17 +1231,17 @@ func (e *historyEngineImpl) DescribeWorkflowExecution(
 				Name: executionInfo.TaskQueue,
 				Kind: enumspb.TASK_QUEUE_KIND_NORMAL,
 			},
-			WorkflowExecutionTimeoutSeconds:   int32(executionInfo.WorkflowExecutionTimeout),
-			WorkflowRunTimeoutSeconds:         int32(executionInfo.WorkflowRunTimeout),
-			DefaultWorkflowTaskTimeoutSeconds: int32(executionInfo.DefaultWorkflowTaskTimeout),
+			WorkflowExecutionTimeout:   timestamp.DurationPtr(time.Duration(executionInfo.WorkflowExecutionTimeout) * time.Second),
```

Comment:
MutableState calls field WorfklowExecutionTimeout and stores as int which is seconds?  I think we should try to fix this next.  @shawnhathaway can you look into this?

Reply:
Yea, there will likely be some intermediate conversions on POGO objects in persistence that will need to be plumbed/removed. Will be looking at this.